org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -311,4 +313,31 @@ public void handleSaved(DidSaveTextDocumentParams params) {
 		}
 	}
 
+	private ICompilationUnit checkPackageDeclaration(String uri, ICompilationUnit unit) {
```

Comment:
could't we perform that check in JDTUtils.getFakeCompilationUnit instead?

Reply:
JDTUtils.getFakeCompilationUnit is called only when creating a file the first time.

Reply:
ok so in JDTUtils .resolveCompilationUnit then

Reply:
I tried, but it didn't work. JDTUtils.getFakeCompilationUnit doesn't use a buffer than read a file from the filesystem.